src/aatoolbox/datasources/chirps/chirps.py

```diff
@@ -0,0 +1,631 @@
+        0.05 or 0.25
+    """
+
+    @abstractmethod
+    def __init__(
+        self,
+        country_config: CountryConfig,
+        geo_bounding_box: GeoBoundingBox,
+        frequency: str,
+        resolution: float = 0.05,
+    ):
+        super().__init__(
+            country_config=country_config,
+            datasource_base_dir="chirps",
+            is_public=True,
+        )
+
+        # round coordinates to correspond with the grid the CHIRPS data is on
+        # non-rounded coordinates can be given to the URL which then
+        # automatically rounds them, but for file saving we prefer to do
+        # this ourselves
+        self._geobb = geo_bounding_box.round_coords(round_val=resolution)
+        self._frequency = frequency
+        self._resolution = resolution
+
+        valid_frequencies = ("daily", "monthly")
+        if self._frequency not in valid_frequencies:
+            raise ValueError(
+                f"The available frequencies are {*valid_frequencies,}"
+            )
+
+        if resolution not in (0.05, 0.25):
+            raise ValueError(
+                f"The given resolution is {self._resolution}, which is "
+                "not available. Has to be 0.05 or 0.25."
+            )
+
+        if self._frequency == "monthly" and self._resolution == 0.25:
+            logger.error(
+                f"{self._frequency}.capitalize() data with a resolution of "
+                "0.25 is not available. Automatically switching to 0.05 "
+                "resolution."
+            )
+            self._resolution = 0.05
+
+    # mypy will give error Signature of "download" incompatible with supertype
+    # "DataSource" due to the arguments not being present in
+    # `DataSource`. This is however valid so ignore mypy
+    def download(
+        self,
+        start_year: int = None,
+        end_year: int = None,
+        start_month: int = None,
+        end_month: int = None,
+        start_day: int = None,
+        end_day: int = None,
+        clobber: bool = False,
+    ):
+        """
+        Download the CHIRPS observed precipitation as NetCDF file.
+
+        Parameters
+        ----------
+        start_year: int, default = None
+            Data will be donwloaded starting from year `start_year`.
+            If None, it is set to 1981.
+        end_year: int, default = None
+            Data will be donwloaded up to year `end_year`.
+            If None, it is set to the year for which most recent data
+            is available.
+        start_month: int, default = None
+            Data will be donwloaded starting from month `start_month`.
+            If None, it is set to January.
+        end_month: int, default = None
+            Data will be donwloaded up to month `end_month`.
+            If None, it is set to the December if the year was specified,
+            otherwise to the month of the year for which most recent data
+            is available.
+        start_day: int, default = None
```

Comment:
I am doubting if we want to give the user this much flexibility or we just always download all days of the month that are available. As it would simplify some things. @turnerm also curious on your thoughts on this.